x-pack/plugins/actions/server/lib/try_catch.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// functional version of try/catch, allows you to not have to use
+// `let` vars initialied to `undefined` to capture the result value
+
+export async function tryCatch<T>(fn: () => Promise<T>): Promise<T | Error> {
```

Comment:
Handy little function to avoid having to `let` define a variable that would need to be set within a `try` / `catch`.

Happy to use some alternative, didn't find anything simple.